src/run/handlers/server.ts

```diff
@@ -128,19 +124,19 @@ export default async (request: Request, context: FutureContext) => {
       return new Response(body || null, response)
     }
 
-    if (context.waitUntil) {
-      context.waitUntil(requestContext.backgroundWorkPromise)
-    }
-
     const keepOpenUntilNextFullyRendered = new TransformStream({
       async flush() {
         // it's important to keep the stream open until the next handler has finished
         await nextHandlerPromise
-        if (!context.waitUntil) {
-          // if waitUntil is not available, we have to keep response stream open until background promises are resolved
-          // to ensure that all background work executes
-          await requestContext.backgroundWorkPromise
-        }
+
+        // Next.js relies on `close` event emitted by response to trigger running callback variant of `next/after`
+        // however @fastly/http-compute-js never actually emits that event - so we have to emit it ourselves,
+        // otherwise Next would never run the callback variant of `next/after`
+        res.emit('close')
```

Comment:
This is to support callback variant of `after()`:
```
after(async () => {
  // this will not start executing immediately - instead this function will be stored to and start executing after response was ~sent
  // make sense to not compete for any resources while working on response and delay executing this function to after that
})
```

This is not needed to support promise variant like so:
```
after(new Promise(resolve => {
  // this starts executing immediately and will ensure it will finish as long 
  // as function timeout is not reached
})
```